src/components/TupleSelect/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { Input, Select } from "antd";
+import { useLocation } from "react-router";
+import Icon from "components/icons/Icon";
+import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
+import { parseQueryString } from "utils";
+
+const { Option } = Select;
+type option = {
+  value: string;
+  displayName: string;
+  placeHolderText: string;
+};
+interface TupleSelectProps {
+  options: option[];
+}
+export const TupleSelect: React.FC<TupleSelectProps> = ({ options }) => {
+  const [input, setInput] = useState("");
+  const [selected, setSelected] = useState(options[0].value);
+  const updateQueryParams = useUpdateURLQueryParams();
+  const { search } = useLocation();
+  const queryParams = parseQueryString(search);
+
+  const onClick = () => {
+    const selectedParams = queryParams[selected] as string[];
+    const updatedParams = upsertQueryParam(selectedParams, input);
+    updateQueryParams({ [selected]: updatedParams });
+    setInput("");
+  };
+  const selectedOption = options.find((o) => o.value === selected);
+  return (
+    <Input.Group compact>
+      <Select
+        style={{ width: "30%" }}
+        value={selected}
+        onChange={(v) => setSelected(v)}
```

Comment:
I think this can be `setSelected`

Reply:
I tried this but ran into some type errors with the antd select component exporting a number type while setSelected expects a string